File: pkg/codegen/testing/utils/testdata/deferred-outputs-pp/dotnet/First.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Random = Pulumi.Random;

namespace Components
{
    public class FirstArgs : global::Pulumi.ResourceArgs
    {
        [Input("passwordLength")]
        public Input<double> PasswordLength { get; set; } = null!;
    }

    public class First : global::Pulumi.ComponentResource
    {
        [Output("petName")]
        public Output<string> PetName { get; private set; }
        [Output("password")]
        public Output<string> Password { get; private set; }
        public First(string name, FirstArgs args, ComponentResourceOptions? opts = null)
            : base("components:index:First", name, args, opts)
        {
            var randomPet = new Random.RandomPet($"{name}-randomPet", new()
            {
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            var randomPassword = new Random.RandomPassword($"{name}-randomPassword", new()
            {
                Length = args.PasswordLength,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            this.PetName = randomPet.Id;
            this.Password = randomPassword.Result;

            this.RegisterOutputs(new Dictionary<string, object?>
            {
                ["petName"] = randomPet.Id,
                ["password"] = randomPassword.Result,
            });
        }
    }
}

File: pkg/codegen/testing/utils/testdata/deferred-outputs-pp/nodejs/first.ts
import * as pulumi from "@pulumi/pulumi";
import * as random from "@pulumi/random";

interface FirstArgs {
    passwordLength: pulumi.Input<number>,
}

export class First extends pulumi.ComponentResource {
    public petName: pulumi.Output<string>;
    public password: pulumi.Output<string>;
    constructor(name: string, args: FirstArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:First", name, args, opts);
        const randomPet = new random.RandomPet(`${name}-randomPet`, {}, {
            parent: this,
        });

        const randomPassword = new random.RandomPassword(`${name}-randomPassword`, {length: args.passwordLength}, {
            parent: this,
        });

        this.petName = randomPet.id;
        this.password = randomPassword.result;
        this.registerOutputs({
            petName: randomPet.id,
            password: randomPassword.result,
        });
    }
}
